File: CatalogService/Models/Book.cs
namespace CatalogService.Models; // Put this entity in the CatalogService.Models namespace.

public class Book // Entity stored in database.
{
    public int Id { get; set; } // Primary key.

    public string Title { get; set; } = ""; // Book title.
    public string Author { get; set; } = ""; // Main author string.
    public string Category { get; set; } = ""; // Category/genre text.
    public string? Description { get; set; } // Optional description.

    public string? Publisher { get; set; } // Optional publisher.
    public decimal? Price { get; set; } // Optional price from CSV.
    public string? PublishMonth { get; set; } // Optional publish month from CSV.
    public int Year { get; set; } // Publish year.

    public string? Isbn { get; set; } // Optional ISBN.
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow; // Audit column for creation time.
}

File: elib_mobile_new/lib/models/book.dart
class Book {
  final int id;
  final String title;
  final String author;
  final String category;
  final int year;
  final String? description;
  final String? publisher;
  final double? price;
  final String? publishMonth;
  final String? isbn;
  final String? createdAtUtc;

  Book({
    required this.id,
    required this.title,
    required this.author,
    required this.category,
    required this.year,
    this.description,
    this.publisher,
    this.price,
    this.publishMonth,
    this.isbn,
    this.createdAtUtc,
  });

  factory Book.fromJson(Map<String, dynamic> json) {
    return Book(
      id: json['id'] as int,
      title: json['title']?.toString() ?? '',
      author: json['author']?.toString() ?? '',
      category: json['category']?.toString() ?? '',
      year: (json['year'] as num?)?.toInt() ?? 0,
      description: json['description']?.toString(),
      publisher: json['publisher']?.toString(),
      price: (json['price'] as num?)?.toDouble(),
      publishMonth: json['publishMonth']?.toString(),
      isbn: json['isbn']?.toString(),
      createdAtUtc: json['createdAtUtc']?.toString(),
    );
  }
}